src/Microsoft.VisualStudio.LanguageServices.Razor/Editor/DefaultVisualStudioDocumentTracker.cs

```diff
@@ -107,19 +110,25 @@ public void Subscribe()
 
                 // We expect the document to have a hierarchy even if it's not a real 'project'.
                 // However the hierarchy can be null when the document is in the process of closing.
-                (hierarchy = _projectService.GetHierarchy(_textBuffer)) != null)
+                (hierarchy = _projectService.GetHierarchy(_textBuffer)) != null &&
+                _textDocumentFactory.TryGetTextDocument(_textBuffer, out var textDocument))
             {
                 projectPath = _projectService.GetProjectPath(hierarchy);
                 isSupportedProject = _projectService.IsSupportedProject(hierarchy);
             }
+            else
+            {
+                return;
+            }
```

Comment:
Why is the else necessary? What case are you handling now that wasn't handled before?

Reply:
totally didn't mean to commit this piece; should have been undone.